src/vignette/http/api_routes.clj

```diff
@@ -0,0 +1,48 @@
+(ns vignette.http.api-routes
+  (:require [cheshire.core :refer :all]
+            [clout.core :refer [route-compile route-matches]]
+            [compojure.core :refer [context routes GET ANY HEAD]]
+            [vignette.http.middleware :refer :all]
+            [vignette.http.route-helpers :refer :all]
+            [vignette.http.proto-routes :as proto]
+            [vignette.util.regex :refer :all]))
+
+(defn create-request-handlers
+  "Creates request handlers for a given route for GET & HEAD using the given route and
+  map generation function."
+  [store route get-handler route->map-fn]
+  ; NOTE: HEAD needs to be first, otherwise compojure will match the GET and set the body to nil
+  [(HEAD route
+         request
+     (handle-head store
+                  (route->map-fn (:route-params request)
+                                 request)))
+   (GET route
+        request
+     (get-handler store
+                  (route->map-fn (:route-params request)
+                                 request)
+                  request))])
+
+(defn api-routes
+  [store]
+  (flatten
+    [(create-request-handlers store proto/scale-to-width-route handle-thumbnail route->thumbnail-auto-height-map)
+     (create-request-handlers store proto/scale-to-width-down-route handle-thumbnail route->thumbnail-auto-height-map)
+     (create-request-handlers store proto/scale-to-height-down-route handle-thumbnail route->thumbnail-auto-width-map)
+     (create-request-handlers store proto/window-crop-route handle-thumbnail route->thumbnail-auto-height-map)
+     (create-request-handlers store proto/window-crop-fixed-route handle-thumbnail route->thumbnail-map)
+     (create-request-handlers store proto/thumbnail-route handle-thumbnail route->thumbnail-map)
+     (create-request-handlers store proto/original-route handle-original route->original-map)]))
+
+(def wiki-context ["/:wikia:image-type/:top-dir/:middle-dir/:original/revision/:revision"
```

Comment:
In src/vignette/setup.clj:25
```(defn image-routes [stores]

  (list
    (def-api-context wiki-context (:wikia-store stores))
    (def-api-context uuid-context (:static-store stores))
    (hlr/legacy-routes (:wikia-store stores))))```

configures the routes for legacy URLs

On Thu, Aug 27, 2015 at 8:08 PM, Nelson Monterroso <notifications@github.com
> wrote:

> In src/vignette/http/api_routes.clj
> <https://github.com/Wikia/vignette/pull/95#discussion_r38126864>:
>
> > +                  (route->map-fn (:route-params request)
> > +                                 request)
> > +                  request))])
> > +
> > +(defn api-routes
> > +  [store]
> > +  (flatten
> > +    [(create-request-handlers store proto/scale-to-width-route handle-thumbnail route->thumbnail-auto-height-map)
> > +     (create-request-handlers store proto/scale-to-width-down-route handle-thumbnail route->thumbnail-auto-height-map)
> > +     (create-request-handlers store proto/scale-to-height-down-route handle-thumbnail route->thumbnail-auto-width-map)
> > +     (create-request-handlers store proto/window-crop-route handle-thumbnail route->thumbnail-auto-height-map)
> > +     (create-request-handlers store proto/window-crop-fixed-route handle-thumbnail route->thumbnail-map)
> > +     (create-request-handlers store proto/thumbnail-route handle-thumbnail route->thumbnail-map)
> > +     (create-request-handlers store proto/original-route handle-original route->original-map)]))
> > +
> > +(def wiki-context ["/:wikia:image-type/:top-dir/:middle-dir/:original/revision/:revision"
>
> is something either in wiki-context or uuid-context? what if it's neither
> (like the legacy thumbnail urls)?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Wikia/vignette/pull/95/files#r38126864>.
>